src/bsoncxx/oid.cpp

```diff
@@ -23,12 +23,7 @@
 namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 
-const oid::init_tag_t oid::init_tag{};
-
-oid::oid() : _is_valid(false) {
-}
-
-oid::oid(init_tag_t) : _is_valid(true) {
+oid::oid() : _is_valid(true) {
```

Comment:
What should happen when you do `oid my_id{"foobarbaz"};`?
Right now (both in this PR and as is), this is fine and just sets `_is_valid` to false. Maybe it should throw? Anyway, in the current state, `_is_valid` is still needed.